src/susa.ts

```diff
@@ -0,0 +1,58 @@
+import { SUSA_MAP, SUSA_CLASSIFIER_MAP } from './constants';
+import { hasProperty } from './utils';
+
+export function susa(num: number, classifier?: boolean): string {
+  validateNumber(num);
+  return classifier ? getClassifierWord(num) : getNumberWord(num);
+}
+
+function getClassifierWord(num: number): string {
+  if (num === 20) {
+    return SUSA_CLASSIFIER_MAP[num];
+  }
+
+  const tens = Math.floor(num / 10) * 10;
+  const ones = num % 10;
+
+  let classifier = '';
+
+  if (hasProperty(SUSA_MAP, tens)) {
+    classifier += SUSA_MAP[tens];
+  }
+
+  if (hasProperty(SUSA_CLASSIFIER_MAP, ones)) {
+    classifier += SUSA_CLASSIFIER_MAP[ones];
+    return classifier;
+  }
+
+  if (hasProperty(SUSA_MAP, ones)) {
+    classifier += SUSA_MAP[ones];
+  }
+
+  return classifier;
+}
+
+function validateNumber(num: number): void {
+  if (Number.isNaN(num) || !Number.isFinite(num) || num <= 0 || num > 100) {
+    throw new Error('지원하지 않는 숫자입니다.');
+  }
+}
+
+function getNumberWord(num: number): string {
+  if (num === 100) {
+    return SUSA_MAP[100];
+  }
+
+  const tens = Math.floor(num / 10) * 10;
+  const ones = num % 10;
+  let result = '';
+
+  if (tens > 0 && hasProperty(SUSA_MAP, tens)) {
+    result += SUSA_MAP[tens];
+  }
+  if (ones > 0 && hasProperty(SUSA_MAP, ones)) {
+    result += SUSA_MAP[ones];
+  }
+
+  return result;
```

Comment:
변할 수 있는 변수 선언인 let보다는 최대한 const를 사용해서, 사이드이펙트를 줄여보는것은 어떤가요?
아래와 같은 방법이 생각나는데, 더 좋은 방법이 있다면 그것도 좋아요!

```suggestion
  const tensWord = tens > 0 && hasProperty(SUSA_MAP, tens) ? SUSA_MAP[tens] : '';
  const onesWord = ones > 0 && hasProperty(SUSA_MAP, ones) ? SUSA_MAP[ones] : '';

  return `${tensWord}${onesWord}`;
```

Reply:
개선 의견 감사드립니다.
저에게 보기 더 좋아보이네요! 🙂

다만, 0보다 큰 조건은 삭제해도 될 것 같다는 의견입니다.

제가 놓친 0보다 커야할 이유가 없다면 다음과 같이 수정하는 것이 어떨까요?

```ts
const tensWord = hasProperty(SUSA_MAP, tens) ? SUSA_MAP[tens] : '';
const onesWord = hasProperty(SUSA_MAP, ones) ? SUSA_MAP[ones] : '';

 return `${tensWord}${onesWord}`;
```

Reply:
https://github.com/toss/es-hangul/pull/201/commits/da227e85bc962f1fa9437fa51460f074e56d4f92 에 반영했습니다.